test/e2e/multikueue/e2e_test.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mke2e
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = ginkgo.Describe("MultiKueue", func() {
+	var (
+		managerNs *corev1.Namespace
+		worker1Ns *corev1.Namespace
+		worker2Ns *corev1.Namespace
+	)
+	ginkgo.BeforeEach(func() {
+		managerNs = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "multikueue-",
+			},
+		}
+		gomega.Expect(k8sManagerClient.Create(ctx, managerNs)).To(gomega.Succeed())
+
+		worker1Ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: managerNs.Name,
+			},
+		}
+		gomega.Expect(k8sWorker1Client.Create(ctx, worker1Ns)).To(gomega.Succeed())
+
+		worker2Ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: managerNs.Name,
+			},
+		}
+		gomega.Expect(k8sWorker2Client.Create(ctx, worker2Ns)).To(gomega.Succeed())
+
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sManagerClient, managerNs)).To(gomega.Succeed())
+		gomega.Expect(util.DeleteNamespace(ctx, k8sWorker1Client, worker1Ns)).To(gomega.Succeed())
+		gomega.Expect(util.DeleteNamespace(ctx, k8sWorker2Client, worker2Ns)).To(gomega.Succeed())
+	})
+	ginkgo.When("Using multiple clusters", func() {
+		ginkgo.DescribeTable("Cluster kubeconfig propagation", func(c *client.Client, key string, ns **corev1.Namespace) {
```

Comment:
Can we have `*corev1.Namespace` instead of `**corev1.Namespace`? And analogously in integration tests.

Reply:
no, https://github.com/kubernetes-sigs/kueue/pull/1360#discussion_r1423509135

Reply:
I think it makes it hard to reason about the test. Maybe we could wrap the body of the test in a regular function, and just call twice for both workers?

Reply:
In particular, I think we we use two `Entry`, then we effectively have two tests, so we run the `BeforeEach` twice. It seems, the intention is to run the `BeforeEach` once, then check kubeconfig propagation for both workers. Unless I'm missing something, I see no point running the `BeforeEach` twice.

Reply:
I can't see how is this hard to read.

We will remove these placeholder test when we add some real ones in 1313.

Reply:
This ** just looks scary, but maybe it's just me :). 

But also that the `BeforeEach` sets up both workers, but in each test we only test one worker, this is counter-intuitive, and wasteful. I would prefer to have a function called twice from within a single `It`, once per worker. This would also eliminate the need for `**` :) 

Anyway, it's not blocking to me, if @alculquicondor and @tenzen-y are happy.

Reply:
Do we even care about different Namespaces for the following PR?

We will probably need the same namespace across all clusters. So I would just remove the namespace altogether here and use `default`.

Reply:
> Do we even care about different Namespaces for the following PR?

Yes, I'd prefer to have every test run in it's own set of namespaces.

Reply:
I guess a dedicated namespace isn't  required, I'm fine either way. I can see we create generated namespaces for regular e2e tests. I guess one benefit could be to reuse your custom cluster for e2e test so that the tests don't conflict with other stuff.

As for the use of `Entry` I still don't see a need for them, and repeating the setup to test per-worker. I would prefer to have the setup done once and check for both workers. E2e tests are generally costly so it seems to make sense to care about running time, espectially if it does not impact (or even improves) readability.

Reply:
This is not the only ginkgo table test in the project.

Reply:
Yeah, I'm not saying "never use table tests", it just feels unnecessary in this context, because it results in repeating the e2e setup per-worker while testing a single worker.

Reply:
https://github.com/kubernetes-sigs/kueue/pull/1360#discussion_r1422042617

Reply:
> Yes, I'd prefer to have every test run in it's own set of namespaces.

But this is running in a different namespace for each cluster. I don't see the point of this test. Even if a placeholder, it should be in a form that will be similar to the incoming test.

Reply:
The namespaces and the ability to use the kubeconfig secret are used in all the new tests that will be added.

Reply:
Let's proceed, given that the follow up PR is already in the pipeline.

/approve
/lgtm